xla/service/gpu/BUILD

```diff
@@ -623,8 +623,11 @@ tsl_gpu_library(
         "//xla/service:collective_ops_utils",
         "//xla/service:global_device_id",
         "//xla/service/llvm_ir:llvm_util",
+        "//xla/stream_executor",
```

Comment:
Could there be missing dependencies in the rocm targets? It seems strange to be adding dependencies to targets you didn't otherwise modify.

Reply:
yes, it's curious why this wasn't needed before actually. Maybe it's pulled in indirectly when on cuda? I could put that in a platform conditional, but that seems weird.

Reply:
I think I've got this resolved a bit better now.